### PR TITLE
[WIP] migrate higlass-python to manzt/hg

### PR DESCRIPTION
**Pull request by manzt**
## Description

Migrates `higlass-python` to `manzt/hg`. This PR adds the `hg` history on top of `higlass-python` with [`--allow-unrealted-histories`](https://jeffkreeftmeijer.com/git-combine/) flag.

I still need to get the documenation up to date (and overhaul the `README.md`) but the hard part is done.

What was changed in this pull request?

- higlass-python is now a Python-only project
- replaces `higlass` with `hg` (under `src/higlass`)
- switches to a `pyproject.toml` config
- uses `setuptools_scm` for versioning
- uses `ruff` instead of `flake8` for linting
- consolidates workflows
- removes all javascript build

rather than installing some package called `hg` the package is `higlass` and the convension for usage in Python is with an alias:

```python
import higlass as hg
```

Why is it necessary?

for the future of higlass use in Python

## Checklist

- [ ] Unit tests added or updated
- [ ] Update `examples.ipynb` notebook
- [ ] Documentation added or updated
- [ ] Updated CHANGELOG.md
- [x] Ran `black` on the root directory

### CODE REVIEW

**Comment by manzt** [comment]
cc: @nvictus

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -e .[dev]
+
+    - name: Formatting with Black
+      run: |
+        black . --check
+
+    - name: Lint with Ruff
```
Comment (pkerpedjiev) [inline .github/workflows/ci.yml:37]: TIL about Ruff 🐶

**Review by pkerpedjiev — COMMENTED** [review]
Took a cursory time-constrained look and here's a few thoughts:

1. The new API is wonderful! Nice work!!
2. The attention to detail in this MR is admirable (love how you updated lines in the LICENSE).
3. I don't know what `server.register` does.
4. More docstrings would be nice, especially for the lock stuff.
5. I'm almost certainly missing something but did the docs change or is that something that will come later?

**Comment by manzt** [comment]
> I don't know what `server.register` does.

At a high level, `server` is a global instance of `HgServer` (probably should rename to `HiGlassServer`). This object is responsible for starting a background _thread_ that exposes various tilesets to the font end. The issue is that we need to encode the Python Tileset (e.g., object that implements `tiles` and `tileset_info`) to the front-end _within_ the schema built. by the core API. This means we need both a `server` and `uuid` to embed in the config (something that we can only get from the "server").

I wanted the server to be pluggable, so non-clodius libraries can implement the `Tileset` Protocol and host their own data sources. So this is usually a two step process with the server:

```python
from higlass.tilesets import cooler

tileset = cooler("./data.mcool") # configures a tileset for a specific file
assert hasattr(tileset, 'tiles') && hasattr(tileset, "tileset_info")

tileset_resource = server.add(tileset)
tileset_resource # has uuid /server info and helper functions to create tracks from this tileset
```

This is kind of annoying both 1.) need to create a tileset (which is likely never used in the code else where) and then 2.) add the tileset to the server manually before being able to use it. So `server.register` just wraps a factory function for a tileset to make the above simplify to:

```python
from higlass import server
import higlass.tilesets

cooler = server.register(higlass.tilesets.cooler) # done inside `__init__.py`

tileset_resource = cooler("./data.mcool")
```

> More docstrings would be nice, especially for the lock stuff.

Yeah, I'll get working on that. Just wanted to get the "bones" of this PR out of the way.

> I'm almost certainly missing something but did the docs change or is that something that will come later?

Docs have not changed yet, and I kept the old ones checked in so that we could somehow version them. I was planning on doing that in a separate PR if that's alright.

**Comment by nvictus** [comment]
> Docs have not changed yet, and I kept the old ones checked in so that we could somehow version them. I was planning on doing that in a separate PR if that's alright.

+1

File: .github/workflows/ci.yml (hunk 1 above)
Comment (manzt) [reply .github/workflows/ci.yml]: oh, also ruff is awesome.

**Pull request merged**